py/day2.py

```diff
@@ -0,0 +1,98 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import ClassVar, Generator
+
+
+class Item:
+    score: ClassVar[int]
+
+    @property
+    def beats(self) -> type[Item]:
+        # can't define this as a class level attribute, classes hasn't been defined yet
+        raise NotImplementedError(
+            f"{self.__class__.__name__} must return the class it beats"
+        )
+
+    def __gt__(self, other: Item) -> bool:
+        return isinstance(other, self.beats)
+
+    def __eq__(self, other: object) -> bool:
+        return self.__class__ == other.__class__
+
+
+class Rock(Item):
+    score = 1
+
+    @property
+    def beats(self):
+        return Scissors
+
+
+class Paper(Item):
+    score = 2
+
+    @property
+    def beats(self):
+        return Rock
+
+
+class Scissors(Item):
+    score = 3
+
+    @property
+    def beats(self):
+        return Paper
+
+
+def rps_factory(char: str) -> Rock | Paper | Scissors:
+    # TODO: pattern matching?
+    class_mapping: dict[str, type[Rock] | type[Paper] | type[Scissors]] = {
+        "A": Rock,
+        "X": Rock,
+        "B": Paper,
+        "Y": Paper,
+        "C": Scissors,
+        "Z": Scissors,
```

Comment:
I removed the `enum` stuff from this `.py` version. It didn't end up being that helpful as this is the only place where it was used.